diff-hl.el

```diff
@@ -141,6 +141,24 @@ the end position as its only argument."
                  (const :tag "Highlight the first column"
                         diff-hl-revert-highlight-first-column)))
 
+(defcustom diff-hl-global-ignore-modes '(not image-mode)
+  "Modes for which `diff-hl-mode' is automagically turned on.
+This is taken directly from `font-lock-global-modes'.
+If nil, means no modes have Font Lock mode automatically turned on.
+If t, all modes that support Font Lock mode have it automatically turned on.
+If a list, it should be a list of `major-mode' symbol names for which Font Lock
+mode should be automatically turned on.  The sense of the list is negated if it
+begins with `not'.  For example:
+ (c-mode c++-mode)
```

Comment:
You can probably use a list `(prog-mode)` as a better example. Nobody's going to enable this is C and C++ only.

Reply:
So the font-lock implementation only looks at concrete modes like C++ and C mode because it just uses `(memq major-mode diff-hl-global-ignore-modes)`. 

I have implemented the described behavior here: `(seq-find (lambda (a) (derived-mode-p a major-mode)) diff-hl-global-modes)`, but the problem is that using `seq-find` requires emacs 25.1 while this package currently support 24.4

Do you think it's ok to bump the requirements up to emacs 25.1? Is anyone still using 24.4? Or should I work around this by defining a custom `seq-find` implementation?

Reply:
> So the font-lock implementation only looks at concrete modes like C++ and C mode because it just uses (memq major-mode diff-hl-global-ignore-modes).

I'm talking about the example in the documentation. Not the implementation.

> I have implemented the described behavior here: (seq-find (lambda (a) (derived-mode-p a major-mode))

This is not actually how `derived-mode-p` is used. You can try this: `(apply #'derived-mode-p diff-hl-global-modes)`.

OTOH, if anyone wanted to disable `diff-hl` in `fundamental-mode` only, perhaps this is not a good idea. Because every major mode is derived from fundamental. Then what `turn-on-font-lock-if-desired` is doing is better. IDK which one to pick, honestly, but in `company-mode-on` I just followed its example.

Reply:
Yeah I agree I think it's probably better to keep it like it is in `company-mode` and `font-lock`. Maybe to make the example more realistic it should be changed to `(not c-mode c++-mode)` for example

Reply:
Perhaps just explain the default value in the documentation (that there's little point is using this mode for images, and it was too slow as well, and the default value avoids it).